src/lib.rs

```diff
@@ -424,6 +424,7 @@ impl_trait!(f32);
 impl_trait!(f64);
 impl_trait!(usize);
 impl_trait!(isize);
+impl_trait!(());
```

Comment:
This should only implement `TypeName` and `ToByteSlice` and maybe `ToMutByteSlice`. Implementing `FromByteSlice` gives something that will only ever panic because of the division, and it also conceptually makes no sense (what length should the returned slice be?).